code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -129,7 +132,7 @@
 	for(var/obj/item/module in added_modules)
 		add_module(module, FALSE, FALSE)
 	for(var/module in held_modules)
-		if(module)
+		if(module in modules)
```

Comment:
when will this check not pass?

Reply:
modules is the newly built set of modules the borg can use, so if they had something in their inventory that isn't usable now, it won't be put back in their inventory. The case of uninstalling the illegal module from a borg while it has the stun prod equipped for example.